libs/common/bit_array.go

```diff
@@ -118,7 +118,11 @@ func (bA *BitArray) Or(o *BitArray) *BitArray {
 		return bA.Copy()
 	}
 	bA.mtx.Lock()
-	defer bA.mtx.Unlock()
+	o.mtx.Lock()
+	defer func() {
+		bA.mtx.Unlock()
+		o.mtx.Unlock()
+	}()
```

Comment:
The reason I didn't bundle the defers is because my understanding is that each defer call is kinda slow: https://github.com/golang/go/issues/14939

Do you think the performance should matter here? Its a bit of a micro-performance tweak, and its _not at all_ clear that it applies here, so I'd be happy to switch to the more easy to maintain version :)

Reply:
As it is to be expected that `BitArray` might be used in tight loops this is a valid concern.